base/v0_3_exp/translate.go

```diff
@@ -387,7 +423,21 @@ func (c Config) addMountUnits(ret *types.Config) translate.TranslationSet {
 			continue
 		}
 		fromPath := path.New("yaml", "storage", "filesystems", i, "with_mount_unit")
-		newUnit := mountUnitFromFS(fs)
+		remote := false
+		// check filesystems targeting /dev/mapper devices against LUKS to determine if a
+		// remote mount is needed
+		if strings.HasPrefix(fs.Device, "/dev/mapper/") || strings.HasPrefix(fs.Device, "/dev/disk/by-id/dm-name-") {
+			for _, luks := range c.Storage.Luks {
+				// LUKS devices are opened with their name specified
+				deviceName := filepath.Base(fs.Device)
+				if deviceName == luks.Name || deviceName == fmt.Sprintf("dm-name-%s", luks.Name) {
```

Comment:
This throws away information (whether to expect a `dm-name-` prefix) and then tries to add it back.  In the _extremely unlikely case_ that someone creates LUKS devices named `foo` and `dm-name-foo`, it will get confused.  Why not flip it around?

```suggestion
				if fs.Device == fmt.Sprintf("/dev/mapper/%s", luks.Name) || fs.Device == fmt.Sprintf("/dev/disk/by-id/dm-name-%s", luks.Name) {
```